src/infi/systray/traybar.py

```diff
@@ -250,6 +277,9 @@ def _command(self, hwnd, msg, wparam, lparam):
     def _execute_menu_option(self, id):
         menu_action = self._menu_actions_by_id[id]
         if menu_action == SysTrayIcon.QUIT:
+            nid = NotifyData(self._hwnd, 0)
+            Shell_NotifyIcon(NIM_DELETE, ctypes.byref(nid))
+            PostQuitMessage(0)  # Terminate the app.

```

Comment:
You're right. It looks like there should be a single function or method that handles all the termination that can be called by `_destroy` and the above `SysTrayIcon.QUIT`. I'll add one. That will also be the right place for the TODO above on releasing resources.